spec/app/tournament/tournament.spec.js

```diff
@@ -0,0 +1,145 @@
+        expect(mTournament.totalMatches()).toEqual(3);
+    });
+
+    it('should have 3 rounds if it has 8 players', function() {
+        var mTournament = new tournament.Tournament();
+        mTournament.addPlayer(new player.Player("Spiderman", 1));
+        mTournament.addPlayer(new player.Player("Batman", 1));
+        mTournament.addPlayer(new player.Player("Ironman", 1));
+        mTournament.addPlayer(new player.Player("Wolverine", 1));
+        mTournament.addPlayer(new player.Player("Thor", 1));
+        mTournament.addPlayer(new player.Player("The Thing", 1));
+        mTournament.addPlayer(new player.Player("Sandman", 1));
+        mTournament.addPlayer(new player.Player("Hiro Nakamura", 1));
+        mTournament.generate();
+        expect(mTournament.totalRounds()).toEqual(3);
+    });
+
+    it('should have a round 2 match with two round 1 parents if it has 4 players', function() {
+        var mTournament = new tournament.Tournament();
+        mTournament.addPlayer(new player.Player("Spiderman", 1));
+        mTournament.addPlayer(new player.Player("Batman", 1));
+        mTournament.addPlayer(new player.Player("Ironman", 1));
+        mTournament.addPlayer(new player.Player("Wolverine", 1));
+        mTournament.generate();
+        expect(mTournament.matches[0].round).toEqual(2);
+        expect(mTournament.matches[0].parents[0].round).toEqual(1);
+        expect(mTournament.matches[0].parents[1].round).toEqual(1);
+        expect(mTournament.matches[0].parents[0].parents.length).toEqual(0);
+        expect(mTournament.matches[0].parents[1].parents.length).toEqual(0);
+    });
+
+    it('should pair up players in all round 1 matches', function() {
+        var mTournament = new tournament.Tournament();
+        mTournament.addPlayer(new player.Player("Spiderman", 1));
+        mTournament.addPlayer(new player.Player("Batman", 1));
+        mTournament.addPlayer(new player.Player("Ironman", 1));
+        mTournament.addPlayer(new player.Player("Wolverine", 1));
+        mTournament.addPlayer(new player.Player("Thor", 1));
+        mTournament.addPlayer(new player.Player("The Thing", 1));
+        mTournament.generate();
+        mTournament.matches.forEach( function(match) {
+            if (match.round == 1) {
+                expect(match.players.length).toEqual(2);
+            } else {
+                expect(match.players.length).toEqual(0);
+            }
+        });
+    });
+
+    it('should have every round 1 match two players after generating', function() {
+        var mTournament = new tournament.Tournament();
+        mTournament.addPlayer(new player.Player("Spiderman", 1));
+        mTournament.addPlayer(new player.Player("Batman", 1));
+        mTournament.addPlayer(new player.Player("Ironman", 1));
+        mTournament.addPlayer(new player.Player("Wolverine", 1));
+        mTournament.addPlayer(new player.Player("Thor", 1));
+        mTournament.addPlayer(new player.Player("The Thing", 1));
```

Comment:
Avoid verbosity. Use a private function in the tests like: createTournament({ playersCount: 6})